### ISSUE TITLE
DL4001 when using multi-stage builds

### ISSUE THREAD

**Issue by wheelq** [hadolint/hadolint#322]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
No issues should be raised

### Actual behavior
`[hadolint] Either use Wget or Curl but not both (DL4001)`

### Steps to reproduce the behavior
Create a Dockerfile with multi-stage build.
In the first image use: `curl` and `wget` in the second image.

#### Dockerfile: 
```
FROM debian:9.8 as go1_builder
LABEL product="gotest1"
ENV GO_VERSION="1.12.5"
WORKDIR /
RUN apt update && \
    apt install -y \
      wget \
      ca-certificates && \
      wget "https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz"

FROM debian:9.8 as go2_builder
LABEL product="gotest2"
ENV GO_VERSION="1.12.5"
WORKDIR /
RUN apt update && \ <--[hadolint] Either use Wget or Curl but not both (DL4001)
    apt install -y \
      ca-certificates \
      curl \
      libcurl4-gnutls-dev && \ <-- when && are removed, DL4001 is not reported anymore
      curl -sSLO "https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz"
```

**Comment by lorenzo** [comment]
How can we reproduce this error?

**Comment by wheelq** [comment]
> How can we reproduce this error?

issue updated with Dockerfile

**Comment by lorenzo** [comment]
Thanks, I will take care of this
